### PR TITLE
Add key information to list endpoints in identity.

### PR DESCRIPTION
**Pull request by jefferai**
Also fixes some bugs from before where we were persisting data that we
should not have been (mount type/path).

### CODE REVIEW

File: vault/identity_store_group_aliases_test.go [hunk 1]
```diff
@@ -8,6 +8,121 @@ import (
...
+		t.Logf("alias info: %#v", info)
+		switch {
+		case info["name"].(string) != "groupalias":
+			t.Fatalf("bad name: %v", info["name"].(string))
+		case info["mount_accessor"].(string) != githubAccessor:
+			t.Fatalf("bad mount_accessor: %v", info["mount_accessor"].(string))
+		}
+	}
+
+	// Now do the same with entity info
```
Comment (vishalnayak) [inline vault/identity_store_group_aliases_test.go]: /`entity info`/`group info`

File: vault/identity_store_group_aliases_test.go [hunk 2]
```diff
@@ -8,6 +8,121 @@ import (
...
+		Path:      "group/id",
+	}
+	resp, err = is.HandleRequest(context.Background(), listReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("err:%v resp:%#v", err, resp)
+	}
+
+	keys = resp.Data["keys"].([]string)
+	if len(keys) != 1 {
+		t.Fatalf("bad: length of entity IDs listed; expected: 1, actual: %d", len(keys))
```
Comment (kalafut) [inline vault/identity_store_group_aliases_test.go]: s/entity/group

File: vault/identity_store_aliases.go [hunk 3]
```diff
@@ -416,15 +421,46 @@ func (i *IdentityStore) pathAliasIDList() framework.OperationFunc {
...
+				aliasInfoEntry["mount_path"] = mi.MountPath
+			} else {
+				mi = mountInfo{}
+				if mountValidationResp := i.core.router.validateMountByAccessor(alias.MountAccessor); mountValidationResp != nil {
+					mi.MountType = mountValidationResp.MountType
+					mi.MountPath = mountValidationResp.MountPath
+					aliasInfoEntry["mount_type"] = mi.MountType
+					aliasInfoEntry["mount_path"] = mi.MountPath
+				}
+				mountAccessorMap[alias.MountAccessor] = mi
```
Comment (vishalnayak) [inline vault/identity_store_aliases.go]: Should this be inside the `if` condition above?

File: vault/identity_store_group_aliases.go [hunk 4]
```diff
@@ -267,15 +268,46 @@ func (i *IdentityStore) pathGroupAliasIDList() framework.OperationFunc {
...
+				entry["mount_path"] = mi.MountPath
+			} else {
+				mi = mountInfo{}
+				if mountValidationResp := i.core.router.validateMountByAccessor(alias.MountAccessor); mountValidationResp != nil {
+					mi.MountType = mountValidationResp.MountType
+					mi.MountPath = mountValidationResp.MountPath
+					entry["mount_type"] = mi.MountType
+					entry["mount_path"] = mi.MountPath
+				}
+				mountAccessorMap[alias.MountAccessor] = mi
```
Comment (vishalnayak) [inline vault/identity_store_group_aliases.go]: Same here

File: vault/identity_store_aliases.go (hunk 3 above)
Comment (jefferai) [reply vault/identity_store_aliases.go]: No, because if `ok`, it's already in the map

**Pull request merged**